### PR TITLE
wifi: Pull nrf_wifi driver restructuring

### PR DESCRIPTION
**Pull request by krish2718**
nRF70 Wi-Fi driver is now restructured, pull these changes to NCS.

### CODE REVIEW

**Review by bjarki-andreasen — APPROVED** [review]
Super cool :)

**Review by kapbh — APPROVED** [review]

File: doc/nrf/drivers/wifi/nrf70_native.rst [hunk 1]
```diff
@@ -82,7 +82,7 @@ The nRF Wi-Fi driver has the following configuration options:
 Kconfig configuration
 =====================
 
-.. options-from-kconfig:: /../../../../../zephyr/drivers/wifi/nrfwifi/Kconfig.nrfwifi
+.. options-from-kconfig:: /../../../../../zephyr/drivers/wifi/nrf_wifi/Kconfig.nrf_wifi
```
Comment (richabp) [inline doc/nrf/drivers/wifi/nrf70_native.rst:85]: The doc build error seems to be due to the folder path which is "drivers/wifi/nrf_wifi/Kconfig.nrfwifi" not "drivers/wifi/nrf_wifi/**Kconfig.nrf_wifi**" as per https://github.com/nrfconnect/sdk-zephyr/pull/2316/files#diff-36828457087e584a38d19121aafd768954fa2b78f98aef837961de7d5fa25402.

Comment (krish2718) [reply doc/nrf/drivers/wifi/nrf70_native.rst:85]: yes,it's fixed locally thanks, didn't do another push to aggregated with any other fixes, so, was waiting for CI to complete.

**Review by sachinthegreen — APPROVED** [review]

**Review by richabp — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by rado17 — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Comment by krish2718** [comment]
There are two twister failure related to `hostap crypto`, one I have resolved, working on best solution for the Enterprise mode `MBEDTLS_PEM_CERTIFICATE_FORMAT` resolution.

**Pull request merged**